rewrite-java/src/main/java/org/openrewrite/java/tree/JavaType.java

```diff
@@ -488,38 +506,40 @@ public boolean deepEquals(@Nullable JavaType type) {
 
         @Getter(AccessLevel.NONE)
         private final int flagsBitMap;
-
         private final FullyQualified declaringType;
+        private final List<JavaType> typeParameters;
+
         private final String name;
         private final Signature genericSignature;
         private final Signature resolvedSignature;
         private final List<String> paramNames;
         private final List<FullyQualified> thrownExceptions;
 
-        private Method(int flagsBitMap, FullyQualified declaringType, String name,
+        private Method(int flagsBitMap, FullyQualified declaringType, @Nullable List<JavaType> typeParameters, String name,
```

Comment:
Collection types best represent the concept of "missing" with an empty collection, rather than `null`. Collections never being nullable makes them more ergonomic. You can iterate and stream empty collections without fear of NPEs. And then there's only one way of representing "missing" for collection types, rather than two. 

I see `@Nullable` collections in a few places in this code review and recommend making all of them not nullable.

Reply:
good point.